cursor_pagination.py

```diff
@@ -103,6 +103,29 @@ def page(self, first=None, last=None, after=None, before=None):
             additional_kwargs['has_next'] = bool(before)
         return CursorPage(items, self, **additional_kwargs)
 
+    def page(self, first=None, last=None, after=None, before=None):
+        qs = self.queryset
+        self._apply_paginator_arguments(qs, first, last, after, before)
+
+        qs = list(qs)
+        items = qs[:page_size]
+        if last is not None:
+            items.reverse()
+        has_additional = len(qs) > len(items)
+
+        return self._get_cursor_page(items, has_additional, first, last, after, before)
+
+    async def apage(self, first=None, last=None, after=None, before=None):
+        qs = self.queryset
+        self._apply_paginator_arguments(qs, first, last, after, before)
+
+        items = await sync_to_async(list)(qs[:page_size])
+        if last is not None:
+            items.reverse()
```

Comment:
I think this is going to happen after the list has been called, right? 

maybe we can do something like this:

```python
@sync_to_async
def _fetch():
    items = qs[:page_size]

    if last is not None:
        items.reverse()

    return list(items) # I think this will is still required 

return await _fetch()
```

Reply:
I actually would recommend avoiding adding the `sync_to_async` and using an `async for`. So, instead of:
```
items = await sync_to_async(list)(qs[:page_size])
```

```
items = []
async item for item in qs[:page_size]:
     items.append(item)
```

While it looks less "pythonic", it will avoid the `sync_to_async` overhead and leverage the asyncio.

Reply:
@rafaelhbarros I think sync to async is required for django, since it doesn't yet allow to run queries in an async context

Reply:
@patrick91 More recent versions of Django ORM do support [async queries](https://docs.djangoproject.com/en/4.2/topics/async/#queries-the-orm)

Reply:
@rafaelhbarros they use sync_to_async under the hood: https://github.com/django/django/blob/8f205acea94e93a463109e08814f78c09307f2b9/django/db/models/query.py#L660-L661

Reply:
Oh that's disappointing. I still recommend using the `async for` in case the django orm ever fixes this, so the update here wouldn't be necessary.

Reply:
`items.reverse()` is actually calling python's `list.reverse`

`qs` is the queryset object, and `items` is the resulting list that gets passed into the CursorPage. I can't find the exact error/line in django's code but as I recall you can't actually call queryset's `reverse()` after the queryset has been sliced

Fixing the `page_size` variable should at least get the tests to pass, and since we're good with the concept of these changes at least, I'll go ahead and add the rest of the tests (didn't want to do them all if we didn't care for this idea anyway)

Reply:
@patrick91 

> TypeError: 'async for' requires an object with __aiter__ method, got QuerySet

Was getting this error so I explicitly added `.aiterator()` to the queryset loop, which is probably a better generic approach anyway since there is some logic in django to avoid the very for loop we were doing
https://github.com/django/django/blob/e2428292abaca4758a7508175d31667fe2dff57c/django/db/models/query.py#L545

Oddly, I do not have this issue in my personal project, and running this project's tests seems to require postgres and a running db so I admittedly have not gotten that set up.

I don't think these tests autorun otherwise I'd try to make sure it all passes before pinging you again! Otherwise ping me and I'll try to fix this again if the current iteration doesn't work.